### PR TITLE
Deployment testing of Harbor on EKS

### PR DESCRIPTION
**Pull request by dirkmueller**
The following changes are included in this PR:
For EKS:
- Install services outside of terraform
- Configure s3 backend for registry artifacts
- Create SPR databases on RDS Postgres
- Configure s3 backend for terraform
- Remove creation of `persistent` storageclass (just use the default instead)
- Set kubernetes version to 1.18
- Change vm size to `t3a.medium` (t3a.small has a limit of 8 containers per node which is not enough)

For AKS and EKS:
- Use selfsinged certificates
- Remove storageclass definition from values as we will use default
- Use ssl-mode `require` for database (AWS signs the certificates for RDS using its own CA)

For the workflow testing SPR 2.1 on AKS/EKS:
- Renamed to `[PC] SUSE Private Registry:2.1`
- Added EKS to the matrix
- Additional changes to support testing SPR on AKS and EKS on the same job

### CODE REVIEW

File: infra/terraform/eks/main.tf [hunk 1]
```diff
@@ -0,0 +1,49 @@
+provider "azurerm" {
```
Comment (jsuchome) [inline infra/terraform/eks/main.tf:1]: azure? Some string replacement is still needed...

File: .github/workflows/eks-registry-2.1.yml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+          ARM_TENANT_ID: ${{ secrets.TF_AZ_ARM_TENANT_ID }}
+        run: |
+          terraform init -backend-config="storage_account_name=ecosystemci" \
+            -backend-config="container_name=tfstate" \
+            -backend-config="access_key=${{ secrets.TF_AZ_BACKEND_ACCESS_KEY }}" \
+            -backend-config="key=codelab.microsoft.tfstate"
+          terraform workspace new ${NAMESPACE}
+          terraform apply -var-file ../../../tests/registry/${{ matrix.test_scenario }}/infra.tfvars \
+            -auto-approve
+          echo "KUBECONFIG=$(realpath ./azurek8s)" >> $GITHUB_ENV
```
Comment (jsuchome) [inline .github/workflows/eks-registry-2.1.yml:63]: I assume the config is not called `azurek8s`

File: infra/terraform/eks/main.tf (hunk 1 above)
Comment (dirkmueller) [reply infra/terraform/eks/main.tf]: yes, I pushed to the wrong origin

File: .github/workflows/eks-registry-2.1.yml (hunk 2 above)
Comment (dirkmueller) [reply .github/workflows/eks-registry-2.1.yml]: fixed

**Comment by flaviodsr** [comment]
Tested in: https://github.com/SUSE/caps-automation/actions/runs/349274064

**Comment by dirkmueller** [comment]
the additional changes from @flaviodsr look okay to me, no major concerns. I can't approve it as I'M the original author of the PR, so github won't let me.

**Review by stefannica — APPROVED** [review]
Didn't look at it closely, but overall, changes look good to me

**Pull request merged**